controllers/provider-vsphere/pkg/apis/vsphere/validation/cloudprofile.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

Comment:
Can we also have validation + tests for the `Infrastructure` and `ControlPlane` resources, please?

Reply:
added test for ControlPlane. InfrastructureConfig is empty currently, i.e. there is nothing to test.